server/src/main/java/org/opensearch/index/store/FileTrackerImp.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * stores all the files added to the filetracker
+ */
+public class FileTrackerImp implements FileTracker {
```

Comment:
Have you looked at `FileTransferTracker` maybe we could reuse?

Reply:
yes, we could definitely try to reuse, for now we wanted to implement this as an independent prototype without changing the existing pieces much.

Reply:
+1 on keeping them independent. @Bukhtawar we are still letting remote store evolve without adding in the remote index specific implementations. Also these changes are going in on a new feature branch. We can look at merging once remote store has launched and enough bake time has passed.